### DISCUSSION TITLE
[Help]: How to Keep Selected Carousel Item Always Center

### DISCUSSION THREAD

**Discussion by itzomen** [davidjerleke/embla-carousel#810] [Help] [answered]
### Summary

I have a carousel that works well but when scrolling to the left, the selected item is not in the centre, how can I always keep the current slide in the centre. I have `align` set to `center` but it does not center the current slide when sliding to the left, as shown below:

#### Scrolling right
![image](https://github.com/davidjerleke/embla-carousel/assets/61752841/272fa47d-ccbd-4d1d-93fc-a2b7b47e7008)

#### Scrolling left
![image](https://github.com/davidjerleke/embla-carousel/assets/61752841/95e27ef6-036a-4106-b10f-f3a59b968df3)

#### Current Fix
I changed the alignment when scrolling left, to bring the current slide to the centre but the transition is not smooth (does not scroll to). I think there should be a better way to do this and any help would be great.
```tsx
  const [emblaThumbsRef, emblaThumbsApi] = useEmblaCarousel(
    {
      containScroll: 'keepSnaps',
      dragFree: true,
      loop: true,
      align: scrollingLeft ? 'start' : 'center',
    },
    [
      WheelGesturesPlugin({
        forceWheelAxis: 'y',
      }),
    ],
  )
```

### If applicable, which variants of Embla Carousel are relevant to this question?

- [ ] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment, marked as answer]
@itzomen a wild guess is to set your options to:

```ts
import useEmblaCarousel from 'embla-carousel-react'

export function EmblaCarousel() {
  const [emblaRef] = useEmblaCarousel({
    containScroll: false,
    align: 'center' // Can be omitted because it's default
  })

  // ...
}
```

[Read about `containScroll` in the docs](https://www.embla-carousel.com/api/options/#containscroll). If that doesn't solve your problem, please share an example of what you want to achieve using a different carousel library.

Best,
David

**Reply by itzomen** [reply]
Hey David,
Thanks for the feedback and your awesome work on the Embla.

I just figured the `align: center` worked as expected. My issue was caused by the fact that I have setup arrow keys navigations and when using the arrow keys I wasn't scrolling to the selected index. The below solved my issue and the current slide is always centered now.
```tsx
  const handleThumbKeyDown = React.useCallback(
    (event: React.KeyboardEvent<HTMLDivElement>) => {
    event.preventDefault()
      if (event.key === 'ArrowLeft') {
        scrollPrevMain()
        scrollPrevThumbs()
      } else if (event.key === 'ArrowRight') {
        scrollNextThumbs()
        scrollNextMain()
      }
     // The following code was absent
      if (!emblaMainApi || !emblaThumbsApi) return
      setSelectedIndex(emblaMainApi.selectedScrollSnap())
      emblaThumbsApi.scrollTo(emblaMainApi.selectedScrollSnap())
    },
    [emblaMainApi, emblaThumbsApi, scrollPrevMain, scrollPrevThumbs, scrollNextThumbs, scrollNextMain],
  )
```

I guess I just needed to sleep over it.
Thanks again!
